libs/traits/src/lib.rs

```diff
@@ -281,56 +285,111 @@ pub trait CurrencyPrice<CurrencyId> {
 	) -> Option<PriceValue<CurrencyId, Self::Rate, Self::Moment>>;
 }
 
+#[derive(Encode, Decode, Clone, PartialEq, Eq, TypeInfo, RuntimeDebug, MaxEncodedLen)]
+pub enum CompoundingSchedule {
+	/// Interest compounds every second
+	Secondly,
+}
+
+#[derive(Encode, Decode, Clone, PartialEq, Eq, TypeInfo, RuntimeDebug, MaxEncodedLen)]
+pub enum InterestRate<Rate> {
```

Comment:
I'm sure you're right, but for my understanding, why?

Reply:
For small types, such as simple enums, numbers, or pointers, making them `Copy` makes it easy to handle them later i.e. in iterators and vectors. As a rule of thumb, anything under 128bits can (should?) be copied. Things bigger can get worse performance because of the bytes copied, and a reference is preferred. For those cases is better to not add `Copy` and let the caller pass it as a reference or explicitly clone if needed.

An important detail is that Copy only can be added if all internal types are already `Copy` before. So, adding a non-copiable field to a `Copy` struct can lead to a big refactor of the code base.

Given said this, I would add `Copy` to `CompoundingSchedule` because it's like a simple mark that adds an information unit. But not sure about adding it to `InterestRate`.